SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m

```diff
@@ -427,6 +429,51 @@ - (void)handleSystemRequestLockScreenIconURL:(SDLOnSystemRequest *)request {
                               }];
 }
 
+- (void)handleSystemRequestHttp:(SDLOnSystemRequest *)request {
+    if (nil == request.bulkData) {
+        // TODO: not sure how we want to handle http requests that don't have bulk data (maybe as GET?)
+        return;
+    }
+    
+    NSError *error = nil;
+    NSDictionary *body = [NSJSONSerialization JSONObjectWithData:request.bulkData options:kNilOptions error:&error];
+    if (error) {
+        NSLog(@"error creating body from bulk data: %@", error.localizedDescription);
+        return;
+    }
+    
+    [self uploadForBodyDataDictionary:body
+                            URLString:request.url
+                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                        NSString *logMessage = nil;
+                        if (error) {
+                            logMessage = [NSString stringWithFormat:@"OnSystemRequest (HTTP response) = ERROR: %@", error];
+                            [SDLDebugTool logInfo:logMessage withType:SDLDebugType_RPC toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+                            return;
+                        }
+                        
+                        if (data == nil || data.length == 0) {
+                            [SDLDebugTool logInfo:@"OnSystemRequest (HTTP response) failure: no data returned" withType:SDLDebugType_RPC toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+                            return;
+                        }
+                        
+                        // Show the HTTP response
+                        [SDLDebugTool logInfo:@"OnSystemRequest (HTTP response)" withType:SDLDebugType_RPC toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+                        
+                        // Create the SystemRequest RPC to send to module.
+                        SDLPutFile *putFile = [[SDLPutFile alloc] init];
```

Comment:
If the data is large (couple MBs), then most likely yes. Otherwise should be fine. HTTP responses usually aren't more than a couple hundred KBs, right?